File: 1-two-sum/1-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var cache = new Dictionary<int,int>();
        for (int i = 0; i < nums.Length; i++) {
          if (cache.ContainsKey(nums[i])) {
            return new int[]{cache[nums[i]], i};
          }
          cache[target - nums[i]] = i;
        }
        return null;
    }
}

File: 1-two-sum/1-two-sum.java
import java.util.HashMap;

class Solution {
  public int[] twoSum(int[] nums, int target) {
    HashMap<Integer, Integer> cache = new HashMap<Integer, Integer>();
    for (int i = 0; i < nums.length; i++) {
      if (cache.containsKey(nums[i])) {
        return new int[] {cache.get(nums[i]), i};
      }
      cache.put(target - nums[i], i);
    }
    return new int[]{};
  }
}